moved/src/state_actor.rs

```diff
@@ -450,6 +460,36 @@ impl<
         (outcome, receipts)
     }
 
+    fn simulate_transaction(&self, request: TransactionRequest) -> TransactionExecutionOutcome {
+        let mut tx = NormalizedEthTransaction::from(request.clone());
+        if let Some(from) = request.from {
+            tx.nonce = quick_get_nonce(&from.to_move_address(), self.state.resolver());
```

Comment:
I think we should only set the nonce if `request.from.is_some()` and `request.nonce.is_none()`. If the user gave us a nonce then we should use it.